### PR TITLE
feat(ledgerjs-cal): add ethereum testnets

### PR DESCRIPTION
**Pull request by adrienlacombe-ledger**
### ❓ Context

- **Impacted projects**: ledgerjs

- **Linked resource(s)**: ledger-live-assets

Add Ethereum testnets to the plugins json

### ✅ Checklist

- [x] **Test coverage**: n/a
- [x] **Atomic delivery**: yes
- [x] **No breaking changes**: no

### 📸 Demo

n/a

### CODE REVIEW

**Comment by adrienlacombe-ledger** [comment]
@gre you are probably the best to review this PR, thank you :beau:

File: libs/ledgerjs/script/crypto-assets-importer/importers/ethereum-plugins.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const { readFileJSON } = require("../utils");
 const mapObject = (obj, fn) => Object.fromEntries(Object.entries(obj).map(fn));
 
 module.exports = {
-  paths: ["dapps/ethereum", "dapps/bsc", "dapps/polygon"],
+  paths: ["dapps/ethereum", "dapps/bsc", "dapps/polygon", "dapps/ethereum_ropsten", "dapps_goerli"],
```
Comment (gre) [inline libs/ledgerjs/script/crypto-assets-importer/importers/ethereum-plugins.js:9]: I don't understand how dapps_goerli can work. I don't see in https://github.com/LedgerHQ/crypto-assets/tree/main/assets

Comment (adrienlacombe-ledger) [reply libs/ledgerjs/script/crypto-assets-importer/importers/ethereum-plugins.js]: thank you @gre I have fixed the path

**Review by gre — COMMENTED** [review]
lgtm but to "test" this, did you manage to run it and make sure the output plugin file is working?

**Comment by adrienlacombe-ledger** [comment]
> lgtm but to "test" this, did you manage to run it and make sure the output plugin file is working?

that's right, the file on the CDN is already updated to take this into account

**Comment by lambertkevin** [comment]
Hey all, remember that we're merging "feat/" branches only inside a release branch after the code freeze 🙏 
cc @gre @adrienlacombe-ledger

**Comment by adrienlacombe-ledger** [comment]
> Hey all, remember that we're merging "feat/" branches only inside a release branch after the code freeze pray cc @gre @adrienlacombe-ledger

@lambertkevin but the `release` branch is taken from `develop` right?

**Comment by lambertkevin** [comment]
> > Hey all, remember that we're merging "feat/" branches only inside a release branch after the code freeze pray cc @gre @adrienlacombe-ledger
> 
> @lambertkevin but the `release` branch is taken from `develop` right?

That's not the point. Yes the release branch is coming from develop, but our gitflow says that develop should always be stable from a dev point of view, so the only things that are allowed to be merged in it are bugfixes or changes without any impact on the project (changing the doc for example). New features are merged in a release branch only to make it the "unstable" environment until QA made its non reg.

**Comment by adrienlacombe-ledger** [comment]
> > > Hey all, remember that we're merging "feat/" branches only inside a release branch after the code freeze pray cc @gre @adrienlacombe-ledger
> > 
> > 
> > @lambertkevin but the `release` branch is taken from `develop` right?
> 
> That's not the point. Yes the release branch is coming from develop, but our gitflow says that develop should always be stable from a dev point of view, so the only things that are allowed to be merged in it are bugfixes or changes without any impact on the project (changing the doc for example). New features are merged in a release branch only to make it the "unstable" environment until QA made its non reg.

Understood, this is a change without any impact, this is not QAed or anything.

**Comment by lambertkevin** [comment]
> > > > Hey all, remember that we're merging "feat/" branches only inside a release branch after the code freeze pray cc @gre @adrienlacombe-ledger
> > > 
> > > 
> > > @lambertkevin but the `release` branch is taken from `develop` right?
> > 
> > 
> > That's not the point. Yes the release branch is coming from develop, but our gitflow says that develop should always be stable from a dev point of view, so the only things that are allowed to be merged in it are bugfixes or changes without any impact on the project (changing the doc for example). New features are merged in a release branch only to make it the "unstable" environment until QA made its non reg.
> 
> Understood, this is a change without any impact, this is not QAed or anything.

Ok then it shouldn't be a `feat/` branch, but a `support/` branch, to notify reviewers that this has no impact 👌

**Comment by adrienlacombe-ledger** [comment]
> > > > > Hey all, remember that we're merging "feat/" branches only inside a release branch after the code freeze pray cc @gre @adrienlacombe-ledger
> > > > 
> > > > 
> > > > @lambertkevin but the `release` branch is taken from `develop` right?
> > > 
> > > 
> > > That's not the point. Yes the release branch is coming from develop, but our gitflow says that develop should always be stable from a dev point of view, so the only things that are allowed to be merged in it are bugfixes or changes without any impact on the project (changing the doc for example). New features are merged in a release branch only to make it the "unstable" environment until QA made its non reg.
> > 
> > 
> > Understood, this is a change without any impact, this is not QAed or anything.
> 
> Ok then it shouldn't be a `feat/` branch, but a `support/` branch, to notify reviewers that this has no impact ok_hand

ah, sorry for that, thank you, noted for next time!

**Comment by gre** [comment]
@lambertkevin it's very unclear what is  a `feat/` and what is a `support/`
the only guideline I see to identify is on doc https://github.com/LedgerHQ/ledger-live/blob/c5da615831c78b1a2603fefa17423718e7aa17ef/CONTRIBUTING.md#branch-naming

It's a one sentence description and open to interpretation.

- feat/ when adding a new feature to the application or library
- bugfix/ when fixing an existing bug
- support/ for any other changes (refactor, tests, improvements, CI…)

I don't see how the present PR can apply to "support" , unless we judge this is an "improvement" but then, everything can be an improvement. Adding a feature is an improvement.

And the fact this drives a completely different git flow is quite critical. I think we will need to improve this, because shipping through develop vs release have big difference on the impact and the consequences, especially if your "feature" need to be built on top of (e.g. here if there is a critical feature that needs that testnet)

**Comment by lambertkevin** [comment]
> @lambertkevin it's still quite unclear to me what is a `feat/` and what is a `support/` the only guideline I see to identify is on doc https://github.com/LedgerHQ/ledger-live/blob/c5da615831c78b1a2603fefa17423718e7aa17ef/CONTRIBUTING.md#branch-naming
> 
> It's a one sentence description and open to interpretation.
> 
> * feat/ when adding a new feature to the application or library
> * bugfix/ when fixing an existing bug
> * support/ for any other changes (refactor, tests, improvements, CI…)
> 
> I don't see how the present PR can apply to "support" , unless we judge this is an "improvement" but then, everything can be an improvement. Adding a feature is an improvement.
> 
> And the fact this drives a completely different git flow is quite critical. I think we will need to improve this, because shipping through develop vs release have big difference on the impact and the consequences, especially if your "feature" need to be built on top of (e.g. here if there is a critical feature that needs that testnet). I have feeling "feature/" is meant to be "stuff that have big impact" and that it makes sense for big work that is getting in, like adding totally new feature in Live, new screens, something that is driven by a big epic basically.. and regardless if it's actually a feature (e.g. **removing libcore**, what would have that been? because it's a big thing, yet strictly speaking it's not really a "feature")

I agree that we need to improve the explanations for each branch type since what felt logical to me and Valentin isn't that obvious to everybody else.

Here, to me, that would be a support branch for 2 reasons combined:
- Yes you're adding a functionnality, but you're adding it to a tool, so nothing that can really impact either all devs or the end user
- It should be very low impact any by adding 2 testnets to this script

So it would be an "improvement", but most of all, it's a modification on a script (which to me fall in the case of the "..." after CI in the description of the support branch)

**Pull request merged**